### DISCUSSION TITLE
"Reduce values by selected filter" Bug

### DISCUSSION THREAD

**Discussion by fredericalpers** [onOffice-Web-Org/oo-wp-plugin#464] [_Discarded pitches]
_in addition to #461_

### Current State
The checkbox for "Reduce values by selected filter" is visible in forms. E.g. "Preferred contact form" (field name: KontaktformBevorzugt), "Kind of property" (field name: objekttyp) and "kind of property

It does not have any functionality.

### Desired State
The checkbox must be removed from forms, as it does not have any functionality in these cases.

![checkbox-2 (1)](https://user-images.githubusercontent.com/88546396/224730017-3d1b4e04-996b-4d3f-9e6d-b632549e928b.png)

**Comment by fredericalpers** [comment]
@dai-eastgate in addition to #461 here are more examples I found before already. Please let me know how long this fix will approximately take.

**Reply by dai-eastgate** [reply]
Yes, I will check it

**Reply by dai-eastgate** [reply]
@fredericalpers I checked it.
I found that the checkbox for "Reduce values by selected filter" is work normal in  "application search form".

https://user-images.githubusercontent.com/106214469/224927219-9b7b037b-b870-48ba-934a-e179b26b2736.mp4


Pls, can you check for me exactly Forms Which the checkbox must be removed?

I think I will need 4 days to research and fix it.

**Reply by fredericalpers** [reply]
The chechbox has to be removed from all forms **_but the "applicant search form"_.** The "applicant search form" is for home owners to find suitable buyers. In this case the real estate agent is supposed to be able to reduce the values to only the available property types.

**Reply by dai-eastgate** [reply]
Thanks for your answer. I will fix it.

**Comment by dai-eastgate** [comment]
@fredericalpers 
1. For the "Reduce values ​​by the selected filter" feature only work on the "applicant search form". We fixed it. You can see the video below. 

https://user-images.githubusercontent.com/106214469/228155819-3990fd19-225b-4be3-ac60-aeb148669c27.mp4


2. But when I go to the Frontend, and click the button search, the result for the filter is wrong: the count of contacts is wrong.
Ex: we find "Type of commercialization: Rent", my contacts list has 2 items has search criteria is "Rent" But you can see 4 items show in screen.

https://user-images.githubusercontent.com/106214469/228156044-97d24915-acd0-4bb2-a7c7-ded9210998af.mp4

I checked the code

![image](https://user-images.githubusercontent.com/106214469/228160652-71d5d714-8ffa-465a-a5c5-81f85147764b.png)

The response API seems to be wrong. It return all contacts Which have not field in search criteria. 
Here is the document of the API:
https://apidoc.onoffice.de/actions/informationen-abfragen/suche-nach-passenden-suchkriterien/

Can you check this response API? And give me your opinion on this matter.

**Reply by fredericalpers** [reply]
Thank you for the fix. 

I think the applicantsearch displays the prospective "buyers" correctly. It will display all customers who want to rent and the 2 other ones who didn't specify their preferred "Type of commercialization". They still might want to rent the apartment/house, even tho they didn't specify.

**Reply by dai-eastgate** [reply]
Merged at https://github.com/onOffice-Web-Org/oo-wp-plugin/pull/467
